### ISSUE TITLE
dismiss message on transition

### ISSUE THREAD

**Issue by lessless** [sir-dunxalot/ember-flash-messages#11]
Hello,
thank you for the work, especially for the documentation. 
Here is one question - is it possible to show message so that it will be shown on the next page and cleared after transition?
For example 
```
// profile/notifications
this.currentModel.save().then(() => {
  this.flashMessage({
    content: 'Saved successfully!', 
    duration: 0, 
    type: 'success',
  });
  this.transitionTo('profile');
}); 
```
will display flash on the only on the profile page

**Comment by sir-dunxalot** [comment]
Hi Yevhenii, thanks for opening the issue. I'm hesitant to make clearing on a transition the default functionality because a lot of the time it's beneficial to do the opposite.

In your case you're saving the model on route one, showing the message, and then transitioning to route two. So if we were to clear the queue on transition the message wouldn't really show at all. Do you mean you only want the message to show on route two until the user navigates and then the message should disappear?

To clear the queue there is a [public method called clear](https://github.com/sir-dunxalot/ember-flash-messages/blob/master/addon/queue.js#L28). However, it needs to be more public (i.e. injected into controllers) so that's one issue I see.

**Comment by sir-dunxalot** [comment]
@lessless The 0.6.0 release (merged in #17) has increased public capabilities in a new service [documented here](https://github.com/sir-dunxalot/ember-flash-messages/wiki/Flash-Message-Queue-Service). This should give you enough to show and hide messages at specific times. Please le me know if you have any questions.
